test/integration/scheduler/scheduler_test.go

```diff
@@ -1217,6 +1217,208 @@ var _ = ginkgo.Describe("Scheduler", func() {
+			util.ExpectPendingWorkloadsMetric(prodCQ, 0, 0)
+			util.ExpectReservingActiveWorkloadsMetric(prodCQ, 1)
+			util.ExpectAdmittedWorkloadsTotalMetric(prodCQ, 1)
+		})
+
+		ginkgo.It("Should schedule workloads borrowing quota from ClusterQueues in the same Cohort", func() {
+			prodCQ = testing.MakeClusterQueue("prod-cq").
+				Cohort("all").
+				ResourceGroup(*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "5", "10", "5").Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodCQ)).Should(gomega.Succeed())
+
+			devCQ = testing.MakeClusterQueue("dev-cq").
+				Cohort("all").
+				ResourceGroup(*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "5", "10", "5").Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, devCQ)).Should(gomega.Succeed())
+
+			prodQueue := testing.MakeLocalQueue("prod-queue", ns.Name).ClusterQueue(prodCQ.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, prodQueue)).Should(gomega.Succeed())
+
+			devQueue := testing.MakeLocalQueue("dev-queue", ns.Name).ClusterQueue(devCQ.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, devQueue)).Should(gomega.Succeed())
+			wl1 := testing.MakeWorkload("wl-1", ns.Name).Queue(prodQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
+			wl2 := testing.MakeWorkload("wl-2", ns.Name).Queue(devQueue.Name).Request(corev1.ResourceCPU, "11").Obj()
+
+			ginkgo.By("Creating two workloads")
+			gomega.Expect(k8sClient.Create(ctx, wl1)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, wl2)).Should(gomega.Succeed())
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wl1, wl2)
+			util.ExpectPendingWorkloadsMetric(prodCQ, 0, 1)
+			util.ExpectPendingWorkloadsMetric(devCQ, 0, 1)
+			util.ExpectReservingActiveWorkloadsMetric(prodCQ, 0)
+			util.ExpectReservingActiveWorkloadsMetric(devCQ, 0)
+			util.ExpectAdmittedWorkloadsTotalMetric(prodCQ, 0)
+			util.ExpectAdmittedWorkloadsTotalMetric(devCQ, 0)
+
+			// Delay cluster queue creation to make sure workloads are in the same
+			// scheduling cycle.
+			testCQ := testing.MakeClusterQueue("test-cq").
+				Cohort("all").
+				ResourceGroup(*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "15", "0", "12").Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, testCQ)).Should(gomega.Succeed())
+			defer func() {
+				gomega.Expect(util.DeleteClusterQueue(ctx, k8sClient, testCQ)).Should(gomega.Succeed())
+			}()
+
+			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, prodCQ.Name, wl1)
+			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, devCQ.Name, wl2)
+			util.ExpectPendingWorkloadsMetric(prodCQ, 0, 0)
+			util.ExpectPendingWorkloadsMetric(devCQ, 0, 0)
+			util.ExpectReservingActiveWorkloadsMetric(prodCQ, 1)
+			util.ExpectReservingActiveWorkloadsMetric(devCQ, 1)
+			util.ExpectAdmittedWorkloadsTotalMetric(prodCQ, 1)
+			util.ExpectAdmittedWorkloadsTotalMetric(devCQ, 1)
+		})
+
+		ginkgo.It("Should try next flavor instead of borrowing", func() {
```

Comment:
Let remove this test since we already have similar one on the above, instead let's add a new test case which will cover the cache usage/requestableQuota. it generally look like:
1. create a workload, it will be admitted.
2. create another workload, unadmitted because of the lendingLimit
3. update the lendingLimit
4. the latter workload admitted.

Or you can combine this with the first testcase.